File: cliente-java/src/main/java/com/clientejava/clienteSession/clientApi/UserAPI.java
package com.clientejava.clienteSession.clientApi;

import java.io.IOException;

import com.clientejava.Channel.TCPChannel;

public class UserAPI extends BaseAPI {

    public UserAPI(TCPChannel tcpChannel) {
        super(tcpChannel);
    }

    public void seek(int offset) throws IOException {
        tcpChannel.writeCommand("seek", offset);
        if (tcpChannel.readDatas().equals("invalid seek position")) {
            throw new IllegalArgumentException("invalid seek position");
        }
    }

    public void stop() throws IOException {
        tcpChannel.writeCommand("stop");
        if (!tcpChannel.readDatas().equals("ok")) {
            throw new AssertionError("Expected 'ok' response");
        }
        tcpChannel.close();
    }

    public void pause() throws IOException {
        tcpChannel.writeCommand("pause");
    }
}

File: client/client_session/client_api/user_api.py
from .base_api import BaseAPI


class UserAPI(BaseAPI):
    def seek(self, offset: int):
        self._tcp_channel.write_command("seek", offset)
        if self._tcp_channel.read_datas() == "invalid seek position":
            raise ValueError("invalid seek position")

    def seek_forward(self):
        self._tcp_channel.write_command("seek_forward")
        if self._tcp_channel.read_datas() == "invalid seek position":
            raise ValueError("invalid seek position")

    def seek_backward(self):
        self._tcp_channel.write_command("seek_backward")
        if self._tcp_channel.read_datas() == "invalid seek position":
            raise ValueError("invalid seek position")

    def stop(self):
        self._tcp_channel.write_command("stop")
        assert self._tcp_channel.read_datas() == "ok"
        self._tcp_channel.close()

    def pause(self):
        self._tcp_channel.write_command("pause")
